### PR TITLE
MOB-456 : integrate abacus validation error with button states

### PR DESCRIPTION
**Pull request by mike-dydx**
## Links (dYdX Internal Use Only)
Linear Ticket: [MOB-456 : integrate abacus validation error with button states](https://linear.app/dydx/issue/MOB-456/integrate-abacus-validation-error-with-button-states)

Figma Design: https://www.figma.com/file/mKevZOfE9nj6MZpiolKYW1/dYdX-%E2%80%BA-Mobile?type=design&node-id=5621-14141&mode=design&t=8620ZeK9v5G9SIV4-4

<br/>

## Description / Intuition
- fixes issue where manually inputting custom size value via keyboard was getting overwritten
- adds inline error display to custom limit price area
- uses validation errors instead of custom error displays

<br/>

## Before/After Screenshots or Videos

| Before | After |
|--------|-------|
| <video src=""> | <video src="https://github.com/dydxprotocol/v4-native-ios/assets/149746839/a4584ecc-ba03-43a2-ba61-d73bd5d10fe6"> |
| <video src=""> | <video src="https://github.com/dydxprotocol/v4-native-ios/assets/149746839/8e362fe3-c232-4768-b2af-b454df963825"> |

<br/>

### Type of Change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring or Technical Debt
- [ ] Documentation update
- [ ] Other (please describe: )

### CODE REVIEW

File: dydx/dydxPresenters/dydxPresenters/_v4/TakeProfitStopLoss/dydxTakeProfitStopLossViewPresenter.swift [hunk 1]
```diff
@@ -171,6 +165,8 @@ private class dydxTakeProfitStopLossViewPresenter: HostedViewPresenter<dydxTakeP
         viewModel?.takeProfitStopLossInputAreaViewModel?.stopLossPriceInputViewModel?.value = dydxFormatter.shared.raw(number: triggerOrdersInput?.stopLossOrder?.price?.triggerPrice?.doubleValue, digits: 2)
         viewModel?.takeProfitStopLossInputAreaViewModel?.lossInputViewModel?.value = dydxFormatter.shared.raw(number: triggerOrdersInput?.stopLossOrder?.price?.usdcDiff?.doubleValue, digits: 2)
 
+        viewModel?.customAmountViewModel?.value = triggerOrdersInput?.size?.doubleValue.magnitude.stringValue
```
Comment (mike-dydx) [inline dydx/dydxPresenters/dydxPresenters/_v4/TakeProfitStopLoss/dydxTakeProfitStopLossViewPresenter.swift:167]: fixes manual input issue mentioned in description

File: dydx/dydxPresenters/dydxPresenters/_v4/TakeProfitStopLoss/dydxTakeProfitStopLossViewPresenter.swift [hunk 2]
```diff
@@ -187,7 +183,8 @@ private class dydxTakeProfitStopLossViewPresenter: HostedViewPresenter<dydxTakeP
         }
 
         if let error = errors.first {
-            viewModel?.submissionReadiness = .fixErrors(cta: error.resources.action?.localized)
+            let errorCta = error.resources.action?.localized ?? DataLocalizer.shared?.localize(path: error.resources.action?.stringKey ?? "", params: nil)
```
Comment (ruixhuang) [inline dydx/dydxPresenters/dydxPresenters/_v4/TakeProfitStopLoss/dydxTakeProfitStopLossViewPresenter.swift]: There is an extension to use:

`
extension ErrorString {
    var localizedString: String? {
        localized ?? DataLocalizer.localize(path: stringKey, params: params?.dictionary)
    }
}
`

File: dydx/dydxPresenters/dydxPresenters/_v4/TakeProfitStopLoss/dydxTakeProfitStopLossViewPresenter.swift [hunk 3]
```diff
@@ -295,6 +292,7 @@ private class dydxTakeProfitStopLossViewPresenter: HostedViewPresenter<dydxTakeP
             AbacusStateManager.shared.triggerOrders(input: $0, type: .stoplossusdcdiff)
         }
         viewModel.customAmountViewModel?.onEdited = {
+            print("mmm: custom amount edited \($0)")
```
Comment (ruixhuang) [inline dydx/dydxPresenters/dydxPresenters/_v4/TakeProfitStopLoss/dydxTakeProfitStopLossViewPresenter.swift]: Delete?

Comment (mike-dydx) [reply dydx/dydxPresenters/dydxPresenters/_v4/TakeProfitStopLoss/dydxTakeProfitStopLossViewPresenter.swift]: i'll do a clean up PR at the end of the project. There's many other print statements i need to get rid of. Omitted anyways d8051ac4f88e897f79c6c21abeeb9533841a8e51

File: dydx/dydxPresenters/dydxPresenters/_v4/TakeProfitStopLoss/dydxTakeProfitStopLossViewPresenter.swift (hunk 2 above)
Comment (mike-dydx) [reply dydx/dydxPresenters/dydxPresenters/_v4/TakeProfitStopLoss/dydxTakeProfitStopLossViewPresenter.swift]: addressed d8051ac4f88e897f79c6c21abeeb9533841a8e51

**Review by ruixhuang — APPROVED** [review]

**Pull request merged**